core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1289,24 +1307,54 @@ public void run() {
             channelLogger.log(ChannelLogLevel.INFO, "Address resolved: {0}", servers);
             haveBackends = true;
           }
-          final Map<String, ?> serviceConfig =
-              config.get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG);
-          if (serviceConfig != null && !serviceConfig.equals(lastServiceConfig)) {
-            channelLogger.log(ChannelLogLevel.INFO, "Service config changed");
-            lastServiceConfig = serviceConfig;
+          final Attributes effectiveConfig;
+          if (discardNameResolverServiceConfig) {
+            channelLogger.log(ChannelLogLevel.INFO, "Service config discarded by channel settings");
+            effectiveConfig = Attributes.newBuilder()
+                .setAll(config)
+                .set(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG, lastServiceConfig)
+                .build();
+          } else {
+            Map<String, ?> effectiveServiceConfig =
+                config.get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG);
+            if (effectiveServiceConfig != null
+                && effectiveServiceConfig.containsKey(DnsNameResolver.SERVICE_CONFIG_ERROR)) {
```

Comment:
@carl-mastrangelo is implementing the error handling in the proper way. Whatever hack you put here for error-handling will be thrown away.  The code as of now will just ignore the update if the config parsing failed, or the resolver didn't return a config. You may want to just keep it that way.

Reply:
So can I just assume the attributes passed here is with either no config or a valid config?

Reply:
Reverted the hack.